modules/nf-core/trust4/main.nf

```diff
@@ -34,12 +36,41 @@ process TRUST4 {
     // separate forward from reverse pairs
     def (forward, reverse) = reads.collate(2).transpose()
     def paired_end_mode = reads && (meta.single_end == false) ? "-1 ${forward[0]} -2 ${reverse[0]}" : ''
+    // read format is optional
+    def readFormat = params.read_format ? "--readFormat ${params.read_format}" : ''
+    // add barcode information if present
+    def barcode = ''
+    if (meta.barcode_read) {
+        if (meta.barcode_read == "R1") {
+            barcode = "--barcode ${forward[0]}"
+        } else if (meta.barcode_read == "R2") {
+            barcode = "--barcode ${reverse[0]}"
+        }
+    }
+    else {
```

Comment:
You have defined barcode to be an empty string, there is no reason to reassign the same value. Same goes for umi_read.